zypp/PurgeKernels.cc

```diff
@@ -163,35 +162,35 @@ namespace zypp {
       return false;
     }
 
-    std::set<sat::Solvable::IdType> removedInThisRun;
-    removedInThisRun.insert( pi.id() );
+    std::set<sat::Solvable> removedInThisRun;
+    removedInThisRun.insert( slv );
 
-    for ( auto it = pool.byStatusBegin( toBeUninstalledFilter ); it != pool.byStatusEnd( toBeUninstalledFilter );  it++  ) {
+    for ( auto p : pool.byStatus( toBeUninstalledFilter ) ) {
```

Comment:
const auto & for range based?

Reply:
That's the downside of `auto`. For a `PoolItem` here `const&` makes sense. In the case below (and several other for loops) it's debatable whether a ` const sat::Solvable &` AKA `const unsigned &` is better than passing the unsigned by value.
Unfortunately `auto` disguises the types rationale. In my code you'd find `const PoolItem &` and `satSolvable` (not even const as it has no not-const methods).